### PR TITLE
feat: add disableSubmodules for git artifacts

### PR DESCRIPTION
**Pull request by zachaller**
This adds the ability to disable submodule for git artifacts we are currently running this. We have a use case where we have a submodule directory that is only needed for local development and to avoid the overhead of auth to another repo during builds it is nice to be able to turn of git submodules on those repos. 

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

**Comment by zachaller** [comment]
@terrytangyuan in regards to https://cloud-native.slack.com/archives/C01QW9QSSSK/p1620238536456600

File: workflow/artifacts/git/git.go [hunk 1]
```diff
@@ -115,9 +116,15 @@ func (g *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 		return err
 	}
 	defer closer()
+
+	var recurseSubmodules = git.DefaultSubmoduleRecursionDepth
+	if inputArtifact.Git.DisableSubmodules {
+		log.Info("Setting recurse submodules to NoRecurseSubmodules")
```
Comment (terrytangyuan) [inline workflow/artifacts/git/git.go:122]: Perhaps more meaningful/readable log here, e.g. "Recursive cloning of submodules is disabled".

**Review by terrytangyuan — APPROVED** [review]
Thanks for your contribution! Just one minor comment. Otherwise LGTM.

File: workflow/artifacts/git/git.go (hunk 1 above)
Comment (zachaller) [reply workflow/artifacts/git/git.go]: Done

**Review by jessesuen — APPROVED** [review]

**Pull request merged**